common/lib/power_status.c

```diff
@@ -0,0 +1,65 @@
+#include "power_status.h"
+
+#include <stdio.h>
+
+#include "hal_gpio.h"
+#include "snoop.h"
+
+static bool is_DC_on = false;
+static bool is_DC_on_delayed = false;
+static bool is_DC_off_delayed = false;
+static bool is_post_complete = false;
+
+void set_DC_status(uint8_t gpio_num)
+{
+	is_DC_on = gpio_get(gpio_num);
+	printf("set dc status %d\n", is_DC_on);
+}
+
+bool get_DC_status()
+{
+	return is_DC_on;
```

Comment:
This is a stale check...does it need to go read the GPIO and return its value instead of a cached value which may be incorrect?

Reply:
Hi @GoldenBug 
Both set_DC_status() and get_DC_status() functions are in the common layer.
The power good pin number is defined in the project code.
The set_DC_status() is called by BIC if the GPIO interrupt is triggered.
In this function, BIC will get GPIO status from the chip.
The other threads in BIC will check host status by calling get_DC_status().